src/types/mod.rs

```diff
@@ -57,6 +73,25 @@ pub struct Time {
     pub nanos: i32,
 }
 
+impl<'a> From<&'a Time> for tendermint::Timestamp {
```

Comment:
Is it confusing to have both `Time` and `Timestamp`? I'm wondering if all of the Amino message could benefit from having `Message` in their name, e.g. `TimeMessage`

Reply:
I like this idea. On the other hand, we don't postfix with `Message` with tendermint either. Our "messages" are usually in a golang `types` package (which I think that's weird / confusing). There was an issue where this was discussed (I think @xla proposed to get rid of those types packages IIRC). If the messages where actually in the packages where they are used instead of one big types package, a `Message` postfix would make more sense.